plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -434,8 +435,12 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       fc::microseconds                _ro_max_trx_time_us{ 0 }; // calculated during option initialization
       ro_trx_queue_t                  _ro_trx_queue;
       std::atomic<uint32_t>            _ro_num_active_trx_exec_tasks{ 0 };
-      std::vector<std::future<bool>>  _ro_trx_exec_tasks_fut;
+      std::condition_variable         _switch_window_cond;
+      std::mutex                      _switch_window_mtx;
+      uint32_t                        _ro_idle_threads{ 0 };
```

Comment:
I think you want your condition variable to be the queue size. So you would use the `ro_trx_queue_t` mutex and not need `_ro_idle_threads`.